src/view/pages/Farmer/tabs/Loan/index.tsx

```diff
@@ -1,78 +1,123 @@
-import { Grid, Row, Content, Button, Text } from 'native-base';
+import { Grid, Row, Content } from 'native-base';
 import CardSummary from '../../../../components/CardSummary';
 import DataTable from '../../../../components/DataTable';
-import styles from './style';
 import Composer from '../../../../hoc/PageComposer';
 
+import { InjectedFabProps } from '../../../../hoc/PageComposer/FabPage';
+import { MapStateToProps, MapDispatchToProps, connect } from 'react-redux';
+import activeRowsActions from '../../../../../store/modules/activeRows/actions';
+import navActions from '../../../../../store/modules/nav/actions';
+import { Route } from '../../../../navigation/routes';
+import { State } from '../../../../../store/types';
+import { dateSort } from '../../../../../utils/DateSort';
+import {
+  getFarmerLoanBalance,
+  getFarmerTotalBalance,
+  getFormattedFarmersTransactions,
+} from '../../../../../store/modules/loan/selectors';
+import styles from './style';
+
 interface OwnPropsType {
-  farmerName: string;
-  totalRemainingBalance: string;
-  totalWeeklyPaymentBalence: string;
-  loanTransactions: any[];
 }
 
 interface DispatchPropsType {
+  navigate(route: Route): void;
+  setActiveLoanEntry(uuid: string): void;
+  navigateToLoanEntry(): void;
 }
 
 interface StorePropsType {
+  farmerLoanBalance: string;
+  farmerTotalBalance: string;
+  loanTransactions: any[];
 }
 
-type PropsType = OwnPropsType & DispatchPropsType & StorePropsType;
-
 interface OwnStateType {
 }
 
-/**
-* Buy Tab Component
-*/
-class Buy extends React.Component<PropsType, OwnStateType> {
+/** Loan NestedPropsType */
+type NestedPropsType = StorePropsType & DispatchPropsType & OwnPropsType;
+
+/** Loan PropsType */
+type PropsType = NestedPropsType & InjectedFabProps;
 
+/** Loan Tab Component */
+class Loan extends React.Component<PropsType, OwnStateType> {
+
+  private onAddPress = () => this.props.navigate(Route.ADD_LOAN_ENTRY);
   private onPressEntry = (uuid: string) => {
     return () => {
-      // example from collect
-      // this.props.setActiveMilkEntry(uuid);
-      // this.props.navigateToMilkEntry();
+      this.props.setActiveLoanEntry(uuid);
+      this.props.navigateToLoanEntry();
     };
   }
+
+  /** React componentDidMount */
+  public componentDidMount() {
+    this.props.listenToFab(this.onAddPress);
+  }
+
   /**
-  * Render method for Buy
+  * Render method for Loan
   */
   public render() {
-    const testData = [{
-      label: 'Total Balance',
-      value: this.props.totalRemainingBalance,
-      units: 'UGX',
-    },                {
-      label: 'Total Weekly Payment',
-      value: this.props.totalWeeklyPaymentBalence,
-      units: 'UGX',
-    },
+    /** A brief summary at the top of the page */
+    const loanDataSummary = [
+      {
+        label: 'Total Loan Balance',
+        value: this.props.farmerLoanBalance,
+        units: 'UGX',
+      },
+      {
+        label: 'Weekly Balance Including Loans',

```

Comment:
This seems to be too long, lets keep it 2-3 word long.

Reply:
I agree, I just couldn't think of a better way to put it. I'll ask for input at the next meeting